lib/backend/test/suite.go

```diff
@@ -100,12 +101,17 @@ func (s *BackendSuite) CRUD(c *check.C) {
 	c.Assert(err, check.IsNil)
 	c.Assert(string(out.Value), check.Equals, string(item.Value))
 
-	// put with binary key and value succeeds
-	key := make([]byte, 1024)
-	rand.Read(key)
+	// put with large key and binary value succeeds.
+	//  NB: DynamoDB has a maximum overall key length of 1024 bytes, so we trim
+	//      our largest keysize to match.
+	keyLen := (1024 / 2) - len(prefix(""))
+	keyBytes := make([]byte, keyLen)
+	rand.Read(keyBytes)
```

Comment:
In tests predictable inputs are usually preferred unless this is fuzzy testing on purpose (which I don't believe here) - can we have a fixed in fixtures instead? Same for data. Would also alleviate the need for hex-formatting.

Reply:
It's an adaptation of the existing key generation code; I think the original intent was to have unique keys for each test invocation so tests can be run in parallel. Pretty much every key used in the test suite is randomly generated to some extent, and, while I agree that the test inputs should be predictable, the scope of the change to make everything behave the same way would swamp the functional change in this patch.

Happy to submit it in another PR, though.